### PR TITLE
Add support for ignore_local_clock

### PR DESCRIPTION
**Pull request by eabecho**
Ignore bad/unreliable local clocks.

### CODE REVIEW

**Comment by eabecho** [comment]
Updated.

File: Modulefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name    'ghoneycutt-ntp'
-version '3.3.0'
+version '3.3.1'
```
Comment (ghoneycutt) [inline Modulefile]: This happened because you did a `git merge` instead of a `git rebase`. This shouldn't be part of the commit history. Luckily, this will not bite us in this situation.

Comment (eabecho) [inline Modulefile]: Indeed that was the case.  I'll use rebase next time.  Thanks for educating me.

**Comment by ghoneycutt** [comment]
This is all good except for the comment about the context names in the spec tests. Please update those and squash the commits and I'll get it merged

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -663,6 +674,14 @@
     end
   end
 
+  context 'with ignore_local_clock' do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: context should have the value such as

``` ruby
context 'with ignore_local_clock set to true' do
```
or 

``` ruby
context 'with ignore_local_clock set to an invalid type' do
```

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -537,6 +537,44 @@
...
+    let(:params) { { :ignore_local_clock => true } }
+    let(:facts) { { :osfamily => 'RedHat' } }
+
+    it { should_not contain_file('ntp_conf').with_content(/^server\s+127.127.1.0/) }
+    it { should_not contain_file('ntp_conf').with_content(/^fudge\s+127.127.1.0 stratum 10/) }
+  end
+
+  context 'with ignore_local_clock (string)' do
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: you can condense these two contexts like this

``` ruby
[true,'true'].each do |value|
  context "with ignore_local_clock set to #{value}" do
    let(:params) { { :ignore_local_clock => value } }
    let(:facts) { { :osfamily => 'RedHat' } }

    it { should contain_file('ntp_conf').without_content(/^server\s+127.127.1.0/) }
    it { should contain_file('ntp_conf').without_content(/^fudge\s+127.127.1.0 stratum 10/) }
  end
end
```

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -537,6 +537,44 @@
...
+    let(:params) { { :ignore_local_clock => 'true' } }
+    let(:facts) { { :osfamily => 'RedHat' } }
+
+    it { should_not contain_file('ntp_conf').with_content(/^server\s+127.127.1.0/) }
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: use `should` and `without.content` instead of `should_not` and `with_content`

**Comment by eabecho** [comment]
Updated.

**Comment by ghoneycutt** [comment]
Released in v3.4.0

Thanks @eabecho !!
